### PR TITLE
BUG: DataFrameGroupBy.quantile raises for non-numeric dtypes rather than dropping columns

### LINKED ISSUE #27892: DataFrameGroupBy.quantile raises for non-numeric dtypes rather than dropping columns
In pandas 0.24.x, we had

```python
In [1]: import pandas as pd

In [2]: pd.DataFrame({"A": ['a', 'b']}).groupby([0, 0]).quantile()
Out[2]:
Empty DataFrame
Columns: []
Index: []
``` 

In 0.25.0, we have

```python
In [3]: pd.DataFrame({"A": ['a', 'b']}).groupby([0, 0]).quantile()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-8152ffc0932b> in <module>
----> 1 pd.DataFrame({"A": ['a', 'b']}).groupby([0, 0]).quantile()

~/sandbox/pandas/pandas/core/groupby/groupby.py in quantile(self, q, interpolation)
   1908             post_processing=post_processor,
   1909             q=q,
-> 1910             interpolation=interpolation,
   1911         )
   1912

~/sandbox/pandas/pandas/core/groupby/groupby.py in _get_cythonized_result(self, how, grouper, aggregate, cython_dtype, needs_values, needs_mask, needs_ngroups, result_is_index, pre_processing, post_processing, **kwargs)
   2236                 vals = obj.values
   2237                 if pre_processing:
-> 2238                     vals, inferences = pre_processing(vals)
   2239                 func = partial(func, vals)
   2240

~/sandbox/pandas/pandas/core/groupby/groupby.py in pre_processor(vals)
   1875             if is_object_dtype(vals):
   1876                 raise TypeError(
-> 1877                     "'quantile' cannot be performed against 'object' dtypes!"
   1878                 )
   1879

TypeError: 'quantile' cannot be performed against 'object' dtypes!
```

This is most relevant for mixed dataframes

```python
In [6]: df = pd.DataFrame({"A": [0, 1], 'B': ['a', 'b']})

In [7]: df.groupby([0, 1]).quantile(0.5)

...

TypeError: 'quantile' cannot be performed against 'object' dtypes!

```

### PR DESCRIPTION
**Pull request by rhshadrach**
- [x] closes #27892
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Unlike what is mentioned in #27892, this will raise if there are no columns to aggregate. Both mean and median raise with "No numeric types to aggregate" in such a case, so I was thinking perhaps we should be consistent with them. Any thoughts @WillAyd and @TomAugspurger?

### CODE REVIEW

**Comment by jreback** [comment]
looks like you are including the changes in #34372 here as well

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -2506,6 +2513,10 @@ def _get_cythonized_result(
             key = base.OutputKey(label=name, position=idx)
             output[key] = result
 
+        # error_msg is "" on an frame/series with no rows or columns
```
Comment (jreback) [inline pandas/core/groupby/groupby.py:2516]: why is this even needed?

File: pandas/tests/groupby/test_quantile.py [hunk 2]
```diff
@@ -232,3 +232,11 @@ def test_groupby_quantile_nullable_array(values, q):
 
     expected = pd.Series(true_quantiles * 2, index=idx, name="b")
     tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize("q", [0.5, [0.0, 0.5, 1.0]])
```
Comment (jreback) [inline pandas/tests/groupby/test_quantile.py:237]: can you also try with an empty frame that has the same dtypes, that should raise?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/groupby/groupby.py (hunk 1 above)
Comment (rhshadrach) [reply pandas/core/groupby/groupby.py:2516]: For `resample.test_base.test_resample_empty_dataframe`. This test explicitly expects the result to be an empty dataframe when the input it empty as well.

File: pandas/tests/groupby/test_quantile.py (hunk 2 above)
Comment (rhshadrach) [reply pandas/tests/groupby/test_quantile.py:237]: It does not raise on an empty frame, but instead returns an empty frame with the same columns of the valid dtypes, in the case of the test `Columns: [b]`.

**Comment by TomAugspurger** [comment]
@rhshadrach merging master should fix the CI failure.

**Comment by rhshadrach** [comment]
@jreback - the response never updated but Travis passed. Responses to your questions are above.

**Comment by rhshadrach** [comment]
Accidentally deleted PR branch

**Review by TomAugspurger — APPROVED** [review]

**Comment by rhshadrach** [comment]
@jreback Friendly ping. Responses to your questions are above.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @rhshadrach

**Pull request merged**